src/pybito.cpp

```diff
@@ -990,9 +1035,17 @@ PYBIND11_MODULE(bito, m) {
            &NNIEngine::SetTPParsimonyDropFilteringScheme,
            "Set filtering scheme to use Top Pruning with Parsimony based on drop from "
            "best score.")
-      .def("set_top_n_score_filtering_scheme", &NNIEngine::SetTopNScoreFilteringScheme,
+      .def("set_top_k_score_filtering_scheme", &NNIEngine::SetTopKScoreFilteringScheme,
            "Set filter scheme that accepts the top N best-scoring NNIs.",
-           py::arg("top_n"), py::arg("max_is_best") = true)
+           py::arg("top_k"), py::arg("max_is_best") = true)
+      // Custom filters
+      .def("set_filter_init_function", &NNIEngine::SetFilterInitFunction)
+      .def("set_filter_pre_score_function", &NNIEngine::SetFilterPreScoreFunction)
+      .def("set_filter_score_loop_function", &NNIEngine::SetFilterScoreLoopFunction)
+      .def("set_filter_post_score_function", &NNIEngine::SetFilterPostScoreFunction)
+      .def("set_filter_evaluate_function", &NNIEngine::SetFilterEvaluateFunction)
+      .def("set_filter_evaluate_loop_function",
+           &NNIEngine::SetFilterEvaluateLoopFunction)
```

Comment:
Do we need to expose the types StaticFilterUpdateFunction, StaticFilterScoreLoopFunction, etc? Or is pybind smart enough to already know them?

Reply:
If you write python functions with the correct number of arguments, pybind can convert them to the C++ std::functions with the correct signature.